Remove mui e2e test

NSeydoux opened the pull request:
This refactors the e2e test app to be just that, meaning it isn't a demo app. As such, it doesn't ned any styling or extra dependency. While being in there, I also converted it to TS for alignment with the rest or our codebase.

There appeared to be unused dependencies as well, removed them too.

NSeydoux committed:
Remove styles and unnecessary dependencies

This is a test app, not a demo one, so it doesn't need any styling, and shouldn't have any cruft.

NSeydoux committed:
Refactor the test app without solid-ui-react

Move some components around, remove all of the styling, and simplify the code going directly to solid-client-authn-browser.

NSeydoux committed:
Convert to TS

For consistency

NSeydoux committed:
Align login sequence with other tests

NSeydoux committed:
Align tags ID with test expectations

NSeydoux committed:
Remove dependency on event

NSeydoux committed:
Uninstall parse-link-header

NSeydoux committed:
Removed file mistakenly added to version control

ThisIsMissEm commented on e2e/browser/testApp/components/accessGrants/index.tsx:
```diff
@@ -0,0 +1,157 @@
+    }
+    (async () => {
+      if(typeof session.info.webId !== "string") {
+        console.error("You must be authenticated to create a resource.")
```
We should probably write this to a log in the viewable test output, instead of just a log message.

NSeydoux replied:
Added an error status to the main ui with 94c0d8df7e83df384763ba2a11b96e955d0efbc6

ThisIsMissEm commented on e2e/browser/testApp/components/accessGrants/index.tsx:
```diff
@@ -0,0 +1,157 @@
+      // Create a file in the resource owner's Pod
+      const resourceOwnerPodAll = await getPodUrlAll(session.info.webId);
+      if (resourceOwnerPodAll.length === 0) {
+        throw new Error(
+          "The Resource Owner WebID Profile is missing a link to at least one Pod root."
+        );
```
Ditto here, just throwing isn't particularly great, and we should report the status in the test UI ideally.

NSeydoux replied:
94c0d8df7e83df384763ba2a11b96e955d0efbc6

ThisIsMissEm commented on e2e/browser/testApp/components/appContent/index.jsx:
```diff
@@ -19,138 +19,101 @@
-      setSharedResourceIri(getSourceUrl(savedFile));
-    })();
-  };
+  const session = getDefaultSession();
```
you can lift this up, as it's a singleton, React need not manage this.

NSeydoux replied:
It's actually not even used, removed it altogether :)

NSeydoux replied:
d7a5432c8eb6e8d030d44dbddaba1180e5c375b3

ThisIsMissEm commented on e2e/browser/testApp/package.json:
```diff
@@ -16,30 +16,16 @@
-    "@inrupt/prism-react-components": "^0.13.1-alpha.2",
     "@inrupt/solid-client": "^1.20.1",
     "@inrupt/solid-client-access-grants": "file:../../../",
     "@inrupt/solid-client-authn-browser": "^1.12.2",
```
```suggestion
    "@inrupt/solid-client-authn-browser": "^1.12.3",
```

Just to save the dependabot PR

NSeydoux replied:
61aaea9d6e70e95ce06380a9f6cc11660934e12b

ThisIsMissEm commented on e2e/browser/testApp/package.json:
```diff
@@ -16,30 +16,16 @@
   },
   "dependencies": {
-    "@inrupt/prism-react-components": "^0.13.1-alpha.2",
     "@inrupt/solid-client": "^1.20.1",
```
```suggestion
    "@inrupt/solid-client": "^1.23.3",
```

NSeydoux replied:
61aaea9d6e70e95ce06380a9f6cc11660934e12b

ThisIsMissEm commented on e2e/browser/testApp/package.json:
```diff
@@ -16,30 +16,16 @@
-      "react": "^16.14.0",
-      "react-dom": "^16.14.0"
-    }
+    "typescript": "^4.9.3"
```
We might be best sticking to `"typescript": "^4.7.4"` until we do the upgrade task, then we can tackle them all at once and know that everything is on the same version of typescript

NSeydoux replied:
e7121f40248b3042775ce554c9d9d18f5611ca32

ThisIsMissEm commented on package.json:
```diff
@@ -131,8 +130,6 @@
     "auth-header": "^1.0.0",
     "base64url": "^3.0.1",
     "cross-fetch": "^3.1.4",
-    "events": "^3.3.0",
-    "parse-link-header": "^2.0.0",
```
Nice catch! I didn't realise we weren't actually not using these

NSeydoux replied:
I'm not even sure why they got there in the first place ^^

ThisIsMissEm reviewed (changes requested):
This looks fantastic, just two outdated dependencies to update, and to revert the typescript 4.9.3 upgrade, as we've a follow up task for that. Plus a minor comment on usage of `getDefaultSession`

I'm okay with the error handling being a follow up task, if you want, though it may make debugging tests in the future harder.

NSeydoux committed:
Update solid-client and authn-browser

NSeydoux committed:
Revert typescipt version

NSeydoux committed:
Removed unused session

NSeydoux committed:
Make everything typescript

NSeydoux committed:
Add error message to test app ui

If an error happens, it is shown in the app UI instead of the console to be easier to spot when debugging

NSeydoux committed:
Enforce linting running on e2e

NSeydoux committed:
fixup! Add error message to test app ui

NSeydoux committed:
fixup! fixup! Add error message to test app ui

NSeydoux committed:
Move from Buffer to Blob

ThisIsMissEm commented on e2e/browser/testApp/components/appContent/index.tsx:
```diff
@@ -0,0 +1,125 @@
+          Log Out
+        </button>
+      </form>
+      <p>
```
```suggestion
      <p data-testid="errorMessage">
```

ThisIsMissEm committed:
Update e2e/browser/testApp/components/appContent/index.tsx

ThisIsMissEm reviewed (approved):
I had one minor change, which I've committed, so it's good to go
